### PR TITLE
refactor: misc quality fixes

### PR DESCRIPTION
**Pull request by anshbansal**
## Checklist

- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

**Comment by anshbansal** [comment]
Working on the `python3` not found error in smoke tests in https://github.com/datahub-project/datahub/pull/7605

File: .github/workflows/qodana-scan.yml [hunk 1]
```diff
@@ -2,15 +2,25 @@ name: Qodana
 on:
   workflow_dispatch:
   pull_request:
+    paths:
```
Comment (aditya-radhakrishnan) [inline .github/workflows/qodana-scan.yml:5]: What does this change do?

**Review by aditya-radhakrishnan — COMMENTED** [review]
These all look mostly okay, can we add some tests for a few of the GraphQL resolvers just to make sure the behavior is the same?

File: .github/workflows/qodana-scan.yml (hunk 1 above)
Comment (anshbansal) [reply .github/workflows/qodana-scan.yml:5]: Only runs this qodana workflow when changes are made for java code

File: datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/resolvers/load/EntityTypeResolver.java [hunk 2]
```diff
@@ -36,12 +36,12 @@ public EntityTypeResolver(
 
 
     private boolean isOnlySelectingIdentityFields(DataFetchingEnvironment environment) {
-        return environment.getField().getSelectionSet().getSelections().stream().filter(selection -> {
+        return environment.getField().getSelectionSet().getSelections().stream().noneMatch(selection -> {
```
[the remark this answers is not in the document]
Comment (anshbansal) [reply datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/resolvers/load/EntityTypeResolver.java:39]: These are standard JDK methods. https://docs.oracle.com/javase/8/docs/api/java/util/stream/Stream.html

`filter` checks `Returns a stream consisting of the elements of this stream that match the given predicate.` then we are counting them to be equal to zero

`noneMatch` doc `Returns whether no elements of this stream match the provided predicate. `

These are essentially the same things. Don't think we should be adding tests for standard JDK methods?

**Pull request closed without merging**